packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1810,11 +1824,15 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
                 if (override.weightedCapacity && Math.floor(override.weightedCapacity) !== override.weightedCapacity) {
                   throw new Error('Weight must be an integer');
                 }
+                if (override.instanceType && override.instanceRequirements) {
```

Comment:
```suggestion
                if (!override.instanceType && !override.instanceRequirements) {
                  throw new Error('You must specify either \'instanceRequirements\' or \'instanceType\'.');
                }
                if (override.instanceType && override.instanceRequirements) {
```
I think we should validate that one of the two properties is specified (will require unit test coverage).